server/src/main/java/io/pravega/schemaregistry/storage/client/WireCommandClient.java

```diff
@@ -0,0 +1,417 @@
+                                                                        final String delegationToken) {
+
+        return getTableUri(tableName).thenCompose(uri -> {
+            final WireCommandType type = WireCommandType.READ_TABLE_ENTRIES;
+            RawClient connection = new RawClient(ModelHelper.encode(uri), connectionPool);
+            final long requestId = connection.getFlow().asLong();
+
+            final IteratorStateImpl token = (state == null) ? IteratorStateImpl.EMPTY : state;
+
+            WireCommands.ReadTableEntries request = new WireCommands.ReadTableEntries(requestId, tableName, delegationToken,
+                    suggestedEntryCount, token.getToken(), Unpooled.EMPTY_BUFFER);
+            return sendRequest(connection, requestId, request)
+                    .thenApply(rpl -> {
+                        handleReply(rpl, connection, tableName, WireCommands.ReadTableEntries.class, type);
+                        WireCommands.TableEntriesRead tableEntriesRead = (WireCommands.TableEntriesRead) rpl;
+                        final IteratorState newState = IteratorStateImpl.fromBytes(tableEntriesRead.getContinuationToken());
+                        final List<TableSegmentEntry> entries =
+                                tableEntriesRead.getEntries().getEntries().stream()
+                                                .map(e -> {
+                                                    WireCommands.TableKey k = e.getKey();
+                                                    return TableSegmentEntry.versioned(k.getData(),
+                                                            e.getValue().getData(),
+                                                            k.getKeyVersion());
+                                                }).collect(Collectors.toList());
+                        return new IteratorItem<>(newState, entries);
+                    });
+        });
+    }
+
+    private WireCommands.TableKey convertToWireCommand(final TableSegmentKey k) {
+        WireCommands.TableKey key;
+        if (k.getVersion() == null) {
+            // unconditional update.
+            key = new WireCommands.TableKey(k.getKey(), WireCommands.TableKey.NO_VERSION);
+        } else {
+            key = new WireCommands.TableKey(k.getKey(), k.getVersion().getSegmentVersion());
+        }
+        return key;
+    }
+
+    private TableSegmentEntry convertFromWireCommand(Map.Entry<WireCommands.TableKey, WireCommands.TableValue> e) {
+        if (e.getKey().getKeyVersion() == WireCommands.TableKey.NOT_EXISTS) {
+            return TableSegmentEntry.notExists(e.getKey().getData(), e.getValue().getData());
+        } else {
+            return TableSegmentEntry.versioned(e.getKey().getData(), e.getValue().getData(), e.getKey().getKeyVersion());
+        }
+    }
+    
+    private void closeConnection(Reply reply, RawClient client) {
+        log.info("Closing connection as a result of receiving: {}", reply);
```

Comment:
Info-level log is likely to result in a lot of logging in production environments. Also, logging the entire reply can cause large amounts of logging. I'd suggest that we: 
* Reduce the logging level to debug or trace. 
* Log the reply at the trace-level only.

Reply:
changed to debug